langchain4j-mcp/src/main/java/dev/langchain4j/mcp/client/DefaultMcpClient.java

```diff
@@ -84,9 +98,15 @@ public String executeTool(ToolExecutionRequest executionRequest) {
         }
         McpCallToolRequest operation =
                 new McpCallToolRequest(idGenerator.getAndIncrement(), executionRequest.name(), arguments);
-        JsonNode jsonNode = transport.executeTool(operation);
+        JsonNode executionResult = null;
+        try {
+            executionResult = transport.executeTool(
+                    operation, toolExecutionTimeout, () -> new CancellationNotification(operation.getId(), "Timeout"));
```

Comment:
This handling will be the same for all transports, so I wanted to avoid code duplication

Reply:
I see, pehaps a good place to put it is in `AbstractMcpTransport`?

Reply:
But yeah it doesn't save much code after all I guess. I'll move it to the transport then

Reply:
> I see, pehaps a good place to put it is in `AbstractMcpTransport`?

I don't think it's worth it, there wouldn't be too much code there, and just complicate things. I moved the CancellationNotification to the transports, and my IDE actually stopped suggesting that there is code duplication, so that was a good idea